recognition/advanced_tree_search.py

```diff
@@ -855,19 +864,20 @@ def cleanup_before_run(self, cmd, retry, *args):
         util.backup_if_exists("build_global_cache.log")
 
     @classmethod
-    def create_config(cls, crp, extra_config, extra_post_config, **kwargs):
-        config, post_config = rasr.build_config_from_mapping(
-            crp,
-            {
-                "lexicon": "speech-recognizer.model-combination.lexicon",
-                "acoustic_model": "speech-recognizer.model-combination.acoustic-model",
-                "language_model": "speech-recognizer.model-combination.lm",
-                "recognizer": "speech-recognizer.recognizer",
-            },
-        )
+    def create_config(cls, crp, search_type, recognizer_type, extra_config, extra_post_config, **kwargs):
+        mapping_dict = {
+            "lexicon": "speech-recognizer.model-combination.lexicon",
+            "acoustic_model": "speech-recognizer.model-combination.acoustic-model",
+            "language_model": "speech-recognizer.model-combination.lm",
+            "recognizer": "speech-recognizer.recognizer",
+            "label_scorer": "speech-recognizer.model-combination.label-scorer",
+        }
+
+        config, post_config = rasr.build_config_from_mapping(crp, mapping_dict)
 
         config.speech_recognizer.recognition_mode = "init-only"
-        config.speech_recognizer.search_type = "advanced-tree-search"
+        config.speech_recognizer.search_type = search_type
+        config.speech_recognizer.reocgnizer_type = recognizer_type
```

Comment:
Do not change config for old setups:
```suggestion
        if recognizer_type != "recognizer":
            config.speech_recognizer.reocgnizer_type = recognizer_type
```